server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -241,18 +243,30 @@ public void onNewClusterState(ClusterState state) {
                                 drainNodesWithDecommissionedAttribute(decommissionRequest);
                             }
                         } else {
-                            // explicitly calling listener.onFailure with NotClusterManagerException as the local node is not leader
-                            // this will ensures that request is retried until cluster manager times out
+                            // since the local node is no longer cluster manager which could've happened due to leader abdication,
+                            // hence retrying the decommission action until it times out
                             logger.info(
-                                "local node is not eligible to process the request, "
-                                    + "throwing NotClusterManagerException to attempt a retry on an eligible node"
+                                "local node is not eligible to process the request, retrying the transport action until it times out"
                             );
-                            listener.onFailure(
-                                new NotClusterManagerException(
-                                    "node ["
-                                        + transportService.getLocalNode().toString()
-                                        + "] not eligible to execute decommission request. Will retry until timeout."
-                                )
+                            decommissionController.retryDecommissionAction(
```

Comment:
This is the transport request which once lands up on any node, will be routed to active cluster manager since its a cluster manager node request. So, yes the logic of retry decommission action will send the transport request to the local which would be old cluster manager and then routed to newly elected cluster manager